# 04 Interpretation


## 1. Data


In [2]:
from pathlib import Path
import pandas as pd

ROOT = Path.cwd().resolve().parent if Path.cwd().name == "notebooks" else Path.cwd().resolve()

TAB_DIR = ROOT / "outputs" / "tables"
SUM_DIR = ROOT / "outputs" / "summary"
SUM_DIR.mkdir(parents=True, exist_ok=True)

descriptive_statistics = pd.read_csv(TAB_DIR / "descriptive_statistics.csv")
anova_results = pd.read_csv(TAB_DIR / "anova_results.csv")
tukey_results = pd.read_csv(TAB_DIR / "tukey_hsd_results.csv")
inference_summary = pd.read_csv(TAB_DIR / "03_inference_summary.csv")

print("Descriptive statistics:")
display(descriptive_statistics)

print("ANOVA results:")
display(anova_results)

print("Tukey HSD results:")
display(tukey_results)

print("Inference summary:")
display(inference_summary)


Descriptive statistics:


,PerceptionOfWeight,WeightPerception,sample_size,mean_bmipct,sd_bmipct,median_bmipct,min_bmipct,max_bmipct
0,1,Very underweight,207,45.503699,35.019935,41.492722,2.250000e-06,99.752151
1,2,Slightly underweight,1436,36.533217,25.777645,31.072561,2.640000e-06,99.798394
2,3,About the right weight,7351,59.713900,24.276332,62.642286,3.720000e-09,99.734701
3,4,Slightly overweight,3318,84.988965,16.426505,90.885855,1.090000e-06,99.939213
4,5,Very overweight,541,92.298578,14.953480,98.176458,6.550834e+00,99.925951


ANOVA results:


,source,sum_sq,df,f_statistic,p_value,alpha,decision
0,C(WeightPerception),"3,176,100.27",4,"1,556.6455",< 0.0001,0.05,Reject H0
1,Residual,"6,553,601.38","12,848",NaN,NaN,0.05,NaN


Tukey HSD results:


,group1,group2,meandiff,p-adj,lower,upper,reject
0,About the right weight,Slightly overweight,25.2751,< 0.0001,23.9864,26.5637,True
1,About the right weight,Slightly underweight,-23.1807,< 0.0001,-24.9584,-21.4030,True
2,About the right weight,Very overweight,32.5847,< 0.0001,29.8399,35.3295,True
3,About the right weight,Very underweight,-14.2102,< 0.0001,-18.5527,-9.8677,True
4,Slightly overweight,Slightly underweight,-48.4557,< 0.0001,-50.4020,-46.5095,True
5,Slightly overweight,Very overweight,7.3096,< 0.0001,4.4527,10.1665,True
6,Slightly overweight,Very underweight,-39.4853,< 0.0001,-43.8994,-35.0711,True
7,Slightly underweight,Very overweight,55.7654,< 0.0001,52.6571,58.8736,True
8,Slightly underweight,Very underweight,8.9705,< 0.0001,4.3896,13.5514,True
9,Very overweight,Very underweight,-46.7949,< 0.0001,-51.8306,-41.7592,True


Inference summary:


,method,group_variable,response_variable,f_statistic,p_value,alpha,decision,post_hoc_test
0,One-way ANOVA,PerceptionOfWeight / WeightPerception,BMIPCT,1556.6455,< 0.0001,0.05,Reject H0,Tukey HSD


## 2. Final Interpretation


The cleaned analysis dataset contained **12,853** students.

The group with the highest mean BMI percentile was **Very overweight**, with mean BMIPCT = **92.30**.

The group with the lowest mean BMI percentile was **Slightly underweight**, with mean BMIPCT = **36.53**.

The one-way ANOVA result was **F = 1556.6455** with **p-value < 0.001**. At alpha = 0.05, the decision is to **Reject H0**.

Because the ANOVA result is statistically significant, Tukey HSD post-hoc comparisons were used. In this analysis, **10 of 10** pairwise comparisons were statistically significant.

Overall, students' perceived weight category is associated with their actual BMI percentile. Students who perceived themselves as overweight generally had higher BMI percentiles than students who perceived themselves as underweight or about the right weight.

This result shows an **association**, not causation, because the YRBS data are observational survey data.


## 3. Limitations


- This is observational survey data, so the result shows association, not causation.
- BMI percentile does not fully capture health status.
- Self-reported survey answers may contain reporting bias.
- Other variables, such as age, sex, physical activity, and nutrition, were not included in this one-way ANOVA.
